src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -36,6 +36,10 @@ public class QueryExecutor<TConnection> : IQueryExecutor
 
         private RetryPolicy _retryPolicy;
 
+        private int? _maxResponseSizeMB;
+        private long? _maxResponseSizeBytes;
```

Comment:
these aren't used in pre-existing code path.. do they need to be nullable? you've calculate what the max value can be, can just set that as default. Then you don't have to have all the explicit casts and `!` null forgiving operators in your code.

Reply:
Fixed.